### ISSUE TITLE
Ongoing issues with Cron

### ISSUE THREAD

**Issue by jerryhops** [turnitin/moodle-plagiarism_turnitin#81]
After applying your latest update to the plagiarism plugin we still have issues with the queue getting stuck. Below is a screen shot of the errors we are seeing:

![tii-cron-errors](https://cloud.githubusercontent.com/assets/16592978/12790873/514c9574-ca9c-11e5-8f1b-4038d8dcafae.PNG)

**Comment by jmcgettrick** [comment]
Hi @jerryhops,
We have fixed this issue but haven't yet released the fix. See: https://github.com/turnitin/moodle-plagiarism_turnitin/blob/develop/lib.php#L2716-L2723. It's occuring because the plugin is attempting to delete a submission that has already been deleted in Turnitin, this is throwing an exception which is causing the cron to get stuck. We'll get this out soon.

**Comment by jerryhops** [comment]
Thanks for that, do you know when this will be released?
Would there be any issue with us applying that patch to the current release?

**Comment by jerryhops** [comment]
One other point, is there anything we can do in Moodle itself to avoid this, for example disable plagiarism on this particular assignment?

**Comment by jmcgettrick** [comment]
You could disable Turnitin for the particular assignment but that would mean no other submissions from that assignment would be sent to Turnitin. 
We don't have a set date to release but there should be no problem just adding the false parameter to the function call on line 2716.

**Comment by jerryhops** [comment]
Thanks for your help.

Do you have any idea when the next release with this fix will be available?
This issue is causing a lot of pain, we have thousands of assignments being submitted and many complaints about missing originality reports / inability to use grademark.

**Comment by jerryhops** [comment]
Hi, I am sorry to pester you but I wondered if you are any closer to putting out another release?

**Comment by jmcgettrick** [comment]
Hi @jerryhops, we had been hoping to release this week but had a couple of minor sticking points. We're hopeful that it'll be early next week at the latest.

**Comment by dwinn** [comment]
Hi @jerryhops, This has now been released.

**Comment by mkpelletier** [comment]
Hello All,
Are you certain this is fixed?  I have just run into something like this again with the present release (2016011105).  I had created an assignment on a course, but the assignment failed to create in Turnitin.  All assignments in other courses were showing pending status. This prevented all subsequent assignments from processing.  After I deleted the offending assignment activity on the course, all the assignments in other courses began to process successfully as shown in the activity log.
